packages/harpguru-core/src/utils/reduce-for-new-column-bounds/reduce-for-new-column-bounds.ts

```diff
@@ -0,0 +1,31 @@
+import type { Dispatch } from 'reactn/default'
+
+import { determineZoomId } from '../determine-zoom-id'
+import { ZoomIds } from '../../types'
+import type { GlobalState } from '../../types'
+
+import { determineNextColumnBounds } from './determine-next-column-bounds'
+
+export const reduceForNewColumnBounds = (
+  global: GlobalState,
+  _dipatch: Dispatch,
+  zoomId: ZoomIds = determineZoomId(global.columnBounds)
+): Pick<GlobalState, 'columnBounds'> => {
+  const {
+    columnBounds,
+    activeHarpStrata: { degreeMatrix: activeDegreeMatrix },
+  } = global
+  const { [0]: exampleHarpRow } = activeDegreeMatrix
+  const { length: activeColumnCount } = exampleHarpRow
+  const newColumnBounds = determineNextColumnBounds(
+    activeColumnCount,
+    columnBounds,
+    zoomId
+  )
+
+  // TODO: read the documentation - if it's the same value as existing then
+  // I think I should be returning an empty object.
```

Comment:
Address this TODO

Reply:
It at least seems likely that we need to try and make the return object's properties match in the `Object.is` comparison that react will use with it's `useReducer` dispatcher: https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-dispatch

I'm not entirely certain but I think that the comparison might actually be done on the entire return object. So if there is no update then we should just return the `global` param.

I should do some testing and review for all of the reducers.

Reply:
Raised as techdebt in #113 

The work done to make the app sizing dynamic (at #107) was used as a platform for additional speculative pull requests to ensure that this was the correct direction to start down.

Rather than addressing the finer details in this or subsequent pull request and potentially struggling to merge up to those subsequent branches I will prefer to record the issues in each merge request and resolve them afterwards.

This has a limit of course. Major issues will not be merged to master. I will need to find a way around that if it happens.